packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -177,12 +252,7 @@ class SizeAnalyzer {
     } else {
       // Print total path and size if currentNode does not have any chilren.
       _printEntitySize(totalPath, byteSize: currentNode.byteSize, level: 2);
-
-      // We picked this file because arm64-v8a is likely the most popular
-      // architecture. Other architecture sizes should be similar.
-      final String libappPath = 'lib/arm64-v8a/$_appFilename';
-      // TODO(peterdjlee): Analyze aot size for all platforms.
-      if (totalPath.contains(libappPath)) {
+      if (totalPath.contains(_locatedAotFilePath.join('/'))) {
```

Comment:
Keep track of the path to the aot file path instead of hard-coding a specific arch